azure-client-runtime/src/main/java/com/microsoft/azure/v2/AzureProxy.java

```diff
@@ -115,6 +202,31 @@ static String getPollUrl(HttpResponse response, String currentPollUrl) {
         return pollUrl;
     }
 
+    private static Observable<HttpResponse> sendPollRequestWithDelay(SwaggerMethodParser methodParser, final HttpClient httpClient, final Value<String> pollUrl, final Value<Long> retryAfterSeconds) {
+        final String fullyQualifiedMethodName = methodParser.fullyQualifiedMethodName();
+
+        return Observable.defer(new Func0<Observable<HttpResponse>>() {
+            @Override
+            public Observable<HttpResponse> call() {
+                final HttpRequest pollRequest = createPollRequest(fullyQualifiedMethodName, pollUrl.get());
+
+                Single<HttpResponse> pollResponse = httpClient.sendRequestAsync(pollRequest);
```

Comment:
here we are taking an assumption that the observable that `httpClient.sendRequestAsync` return is cold so that applying delay later in this observable ensure actual request is really delayed.

I think we could cover both cold and hot by applying delay operator after defer, something like:

```java
return Observable.defer(new Func0<Observable<HttpResponse>>() {
   @Override
   public Observable<HttpResponse> call() {
       final HttpRequest pollRequest = createPollRequest(fullyQualifiedMethodName, pollUrl.get());
       Single<HttpResponse> pollResponse = httpClient.sendRequestAsync(pollRequest);
       return pollResponse.toObservable();
  }
})
.delay(retryAfterSeconds.get() != null ? retryAfterSeconds.get() : 0, TimeUnit.SECONDS);
```

What do you think?

Reply:
Sure, that sounds fine. I'll create a pull request for that.

Reply:
I started implementing this, but then I realized that this won't work as planned. The delay call must be inside the defer() because it needs to re-evaluated for each poll request since retryAfterSeconds may have been updated since the previous poll request. If the delay() is done outside of the defer(), then the delay will be calculated only once and then will be reapplied to every poll request, but the amount of delay will never be updated, even if retryAfterSeconds changes.
In short, I think the delay needs to stay in the defer().

Reply:
@daschult .. sorry, you are right, the current version of delay will not reevaluate the interval. 

Thinking more about this, still feel that relying on the assumption that observable always cold is wrong. What if we revert the logic, i.e. we calculate the delay inside defer and make actual http request outside.. like:

```java
Observable.defer(() -> Observable.just(true).delay(retryAfterSeconds.get() != null ? retryAfterSeconds.get() : 0, TimeUnit.SECONDS))
.flatMap(b -> {
    final HttpRequest pollRequest = createPollRequest(fullyQualifiedMethodName, pollUrl.get());
    Single<HttpResponse> pollResponse = httpClient.sendRequestAsync(pollRequest);
    return pollResponse.toObservable();
});
```

Reply:
We're talking about the observable that performs a polling request, right? It's important that such an observable performs a request when it's subscribed to, each time it's subscribed to, and not at other times.

Reply:
Which one is correct?

1. Perform request as soon as we subscribe and delay the response for the given timespan
2. delay the request for the given timespan and then send request and return response.

The expectation is - method `sendPollRequestWithDelay` is for sending a SINGLE poll request with delay. Above code basically work as expected right?.. when the caller subscribed to the downstream observable, the code inside the defer will run which produce a delay and when the delay done then the request will be send..

Reply:
I think it should be 2. and it looks like you're right.

Reply:
Yes, we should delay, send the request, and immediately emit/return the response. I'll see if I can come up with something for this in a pull request.

Reply:
This should do the trick (delay before request, reevaluate delay each time subscription happens).. 

```java
Observable.defer(() -> Observable.just(true).delay(retryAfterSeconds.get() != null ? retryAfterSeconds.get() : 0, TimeUnit.SECONDS))
.flatMap(b -> {
    final HttpRequest pollRequest = createPollRequest(fullyQualifiedMethodName, pollUrl.get());
    Single<HttpResponse> pollResponse = httpClient.sendRequestAsync(pollRequest);
    return pollResponse.toObservable();
});
```